.github/workflows/ci.yml

```diff
@@ -170,3 +177,57 @@ jobs:
     - uses: codecov/codecov-action@v3
       with:
         token: ${{secrets.CODECOV_TOKEN}}
+
+  go-sanity:
+    name: "Go: Code sanity"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install dependencies
+        run: |
+          sudo DEBIAN_FRONTEND=noninteractive apt update
+          sudo DEBIAN_FRONTEND=noninteractive apt install -y ${{ env.apt_deps }}
+      - uses: actions/checkout@v4
+      - name: Go code sanity check
+        uses: canonical/desktop-engineering/gh-actions/go/code-sanity@main
+        with:
+          golangci-lint-configfile: ".golangci.yaml"
+          tools-directory: "./provd/tools"
+          working-directory: "./provd"
+      - name: Build cmd/provd
+        run: |
+          set -eu
+          cd provd
+          go build ./cmd/provd
+
+  go-tests:
+    name: "Go: Tests"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version-file: "./provd/go.mod"
+
+      - name: Install dependencies
+        run: |
+          sudo DEBIAN_FRONTEND=noninteractive apt update
```

Comment:
we don’t need `DEBIAN_FRONTEND=noninteractive` on apt update (and yes, I’m sure this is a copy from another place which could be fixed :p)